lib/active_record_shards/default_replica_patches.rb

```diff
@@ -161,6 +181,20 @@ def model.on_replica_by_default?
       end
     end
 
+    module Rails42AssociationsAssociationPatch
```

Comment:
```suggestion
    module RailsAssociationsAssociationPatch
      # Not necessary as of Rails 5.2 and beyond
```

What do you think about renaming this given that we're using the patch for Rails 4.2, 5.0 and 5.1 ?